File: src/lib/src/main/java/work/lclpnet/ap2/api/game/EliminationController.java
package work.lclpnet.ap2.api.game;

import net.minecraft.server.level.ServerPlayer;
import net.minecraft.world.damagesource.DamageSource;
import org.jetbrains.annotations.Nullable;
import work.lclpnet.kibu.translate.text.TranslatedText;

public interface EliminationController {

    void eliminateAll(Iterable<? extends ServerPlayer> players);

    void eliminate(ServerPlayer player, @Nullable DamageSource source, @Nullable TranslatedText customMsg);

    default void eliminate(ServerPlayer player) {
        eliminate(player, null, null);
    }

    default void eliminate(ServerPlayer player, @Nullable DamageSource source) {
        eliminate(player, source, null);
    }

    default void eliminate(ServerPlayer player, @Nullable TranslatedText customMsg) {
        eliminate(player, null, customMsg);
    }
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/game/util/EliminationController.kt
package work.lclpnet.ap2.game.util

import net.minecraft.server.level.ServerPlayer
import net.minecraft.world.damagesource.DamageSource
import work.lclpnet.kibu.translate.text.TranslatedText

interface EliminationController {

    fun eliminateAll(players: Iterable<ServerPlayer>)

    fun eliminate(
        player: ServerPlayer,
        source: DamageSource? = null,
        customMsg: TranslatedText? = null,
    )
}
